src/app/core/store-url/store-url.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, Params, Router, RoutesRecognized } from '@angular/router';
+import { filter, pairwise } from 'rxjs';
+import { CategoryRoutingName } from '@app/shared/models';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StoreUrlService {
+  constructor(private router: Router, private route: ActivatedRoute) {}
+
+  setPreviousUrl(): void {
+    this.router.events
+      .pipe(
+        filter((evt: any) => evt instanceof RoutesRecognized),
+        pairwise()
+      )
+      .subscribe((events: RoutesRecognized[]) => {
+        if (events[0].urlAfterRedirects.includes('?')) {
+          localStorage.setItem('pomagamukrainie-url', events[0].urlAfterRedirects.split('?')[0]);
+          localStorage.setItem('pomagamukrainie-query', events[0].urlAfterRedirects.split('?')[1]);
```

Comment:
do poprawy w następnej PR:
pomagamukrainie-url
pomagamukrainie-query

to możesz dać do enum'a, aby nie zrobić przypadkiem literówki